tests/integration/provider/provider_test.go

```diff
@@ -83,5 +83,12 @@ func TestProvider(t *testing.T) {
 			assert.NotEqual(t, providerNsName.(string), namespacedPodNamespace.(string))
 			assert.Equal(t, ns2Name.(string), namespacedPodNamespace.(string))
 		},
+		EditDirs: []integration.EditDir{
+			{
+				Dir:           "step2",
+				Additive:      false,
+				ExpectFailure: true,
```

Comment:
Is this test really testing anything?  I assume it passed previously as well - just with a different error?

Is it worth having a test that *doesn't* fail after these changes?

Reply:
Yeah, good point. I think what I really want to test is that a preview succeeds even when a cluster isn't reachable, but I don't think our test framework currently handles that case (preview only, no update).

Reply:
Justin informed me that this is possible using `RunCommand("pulumi", "preview")`, so I'll update the test accordingly.

Reply:
This turned out to be more difficult than I expected, so I'm opening an issue to update the test framework and will follow up on this later.

Reply:
https://github.com/pulumi/pulumi/issues/3762